dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -1048,3 +1074,102 @@ async fn cmd_nexus_sled_add(
     eprintln!("added sled {} ({})", args.serial, args.part);
     Ok(())
 }
+
+/// Runs `omdb nexus sleds expunge`
+async fn cmd_nexus_sled_expunge(
+    client: &nexus_client::Client,
+    args: &SledExpungeArgs,
+    omdb: &Omdb,
+    log: &slog::Logger,
+) -> Result<(), anyhow::Error> {
+    // This is an extremely dangerous and irreversible operation. We put a
+    // couple of safeguards in place to ensure this cannot be called without
+    // due consideration:
+    //
+    // 1. We'll require manual input on stdin to confirm the sled to be removed
+    // 2. We'll warn sternly if the sled-to-be-expunged is still present in the
+    //    most recent inventory collection
+    use nexus_db_model::Sled;
+    use nexus_db_queries::context::OpContext;
+
+    // First, we need to look up the sled so we know it's serial number.
+    let datastore = args.db_url_opts.connect(omdb, log).await?;
+    let opctx = OpContext::for_tests(log.clone(), datastore.clone());
+    let opctx = &opctx;
+
+    let sled = {
+        use async_bb8_diesel::AsyncRunQueryDsl;
+        use diesel::ExpressionMethods;
+        use diesel::QueryDsl;
+        use diesel::SelectableHelper;
+        use nexus_db_queries::db::schema::sled::dsl;
+
+        let conn = datastore.pool_connection_for_tests().await?;
+        dsl::sled
+            .filter(dsl::id.eq(args.sled_id))
+            .select(Sled::as_select())
+            .first_async::<Sled>(&*conn)
+            .await
+            .with_context(|| format!("failed to find sled {}", args.sled_id))?
+    };
+
+    // Now check whether its sled-agent or SP were found in the most recent
+    // inventory collection.
+    match datastore
+        .inventory_get_latest_collection(opctx)
+        .await
+        .context("loading latest collection")?
+    {
+        Some(collection) => {
+            let sled_present =
+                collection.sled_agents.contains_key(&args.sled_id)
+                    || collection.sps.keys().any(|baseboard| {
+                        baseboard.part_number == sled.part_number()
+                            && baseboard.serial_number == sled.serial_number()
+                    });
```

Comment:
May be worth making this a method on `Collection` since it seems generally useful. I was looking for something like this as part of #5211 myself.

Reply:
Ooh, `Collection` already interns baseboard IDs and stores them in `collection.baseboards`. I can just check that. (If you need something more than "does it exist", I agree adding a method to `Collection` seems reasonable.)